### DISCUSSION TITLE
Helper function to parseADA to Lovelace

### DISCUSSION THREAD

**Discussion by nstanford5** [input-output-hk/marlowe-ts-sdk#143] [Ideas]
Most DApps will present users with the ability to input token amounts to a UI component. Users typically think in terms of ADA amounts, but we need to send Lovelace amounts to the blockchain. 

Should the SDK include a simple helper function to `parseADA` to Lovelace?

Something like..

`const parseADA = (num: number) => { return num * 1000000; };`

On the other side of that, when querying the blockchain for amounts -- it will return Lovelace that needs to be formatted to ADA to present to users. 

Should we also include a `formatLovelace` function?

**Comment by hrajchert** [comment]
I would like to have this type of functionality in a broader concept called TemplateParameters. Currently, that is part of Marlowe Extended, you can check it on the Playground's metadata tab

![Screenshot 2023-12-21 at 23 32 57](https://github.com/input-output-hk/marlowe-ts-sdk/assets/2634059/674a6a43-f17b-43b8-8e1d-5aca87378e3e)

I think it would be good to separate the concept from Marlowe Extended and make a sort DSL with constraints, to be able to say stuff like `amount > 24`, `amount1 <= amount2`, `timeout1 in_between [timeout2 timeout3]`, etc.
